vm/src/org/mozartoz/truffle/nodes/local/BindVarValueNode.java

```diff
@@ -19,30 +20,48 @@ public static BindVarValueNode create() {
 
 	@Specialization(guards = "var.getNext() == var")
 	Object bind1(Variable var, Object value) {
+		print_nlinks(var);
 		var.setInternalValue(value, var);
 		return value;
 	}
 
 	@Specialization(guards = { "var.getNext() != var", "var.getNext().getNext() == var" })
 	Object bind2(Variable var, Object value) {
+		print_nlinks(var);
+		Variable next = var.getNext();
 		var.setInternalValue(value, var);
-		var.getNext().setInternalValue(value, var);
+		next.setInternalValue(value, var);
 		return value;
 	}
 
 	@Specialization(guards = { "var.getNext() != var", "var.getNext().getNext().getNext() == var" })
 	Object bind3(Variable var, Object value) {
+		print_nlinks(var);
+		Variable next1 = var.getNext();
 		var.setInternalValue(value, var);
-		Variable next = var.getNext();
-		next.setInternalValue(value, var);
-		next.getNext().setInternalValue(value, var);
+		Variable next2 = next1.getNext();
+		next1.setInternalValue(value, var);
+		next2.setInternalValue(value, var);
 		return value;
 	}
 
 	@Specialization(contains = { "bind1", "bind2", "bind3" })
 	Object bindLeft(OzVar var, Object value) {
+		print_nlinks(var);
 		var.bind(value);
 		return value;
 	}
 
+	public static void print_nlinks(Variable var) {
+		if (Options.PRINT_NLINKS) {
+			int count = 1;
+			Variable current = var.getNext();
+			while (current != var) {
+				count += 1;
+				current = current.getNext();
+			}
```

Comment:
I think this code should move to Variable, so the iteration logic is encapsulated there.